xdsl/irdl.py

```diff
@@ -613,7 +613,7 @@ def get_variadic_sizes(op: Operation, op_def: OpDef,
                      if isinstance(arg_def, VariadicDef)]
 
     # If the size is in the attributes, fetch it
-    if (attribute_option is not None) and (attribute_option in op_def.options):
```

Comment:
Doesn't this change behavior slightly if attribute_option is actually None and there is a None in op_def.options? I am unsure whether this case can even occur, but I wanted to point this out. Maybe @math-fehr has an intuition?

Reply:
IFF what you say is true, I would suggest that we write a test for this! We need a counter-example to prove that this may not always work. Would you mind contributing/committing a test in this PR that :
> attribute_option is actually None and there is a None in op_def.options?

Reply:
option cannot have `None`, since it is defined as `options: list[IRDLOption] = field(default_factory=list)`.
But overall, I find it very risky to make those changes, even though they are correct.
They make the code more complex to understand, since they hide the corner cases, and these corner cases might arrive later on after a refactor (this makes refactor more difficult as well).

Reply:
reverted